src/form-engine.component.tsx

```diff
@@ -294,6 +296,7 @@ const FormEngine: React.FC<FormProps> = ({
                         patient={patient}
                         formSessionDate={formSessionDate}
                         provider={currentProvider}
+                        role={encounterRole?.uuid}
```

Comment:
Wouldn't it have been easier to fetch the encounter role from the `encounter-form-component` since its used there?

Reply:
Yah, I think that might be better.

Reply:
Actually, for better initialization, I need to define the state in the form-engine, I tried defining it in the EncounterForm but it was undefined